tools/smarterr-parser/main.go

```diff
@@ -0,0 +1,299 @@
+		if regex.MatchString(line) {
+			return patternName
+		}
+	}
+
+	return ""
+}
+
+func printResults(results []FileResult) {
+	if len(results) == 0 {
+		fmt.Println("No non-smarterr error patterns found in resource/datasource files.")
+		return
+	}
+
+	fmt.Printf("Found non-smarterr error patterns in %d files:\n\n", len(results))
+
+	for _, result := range results {
+		for _, errorMatch := range result.Errors {
+			fmt.Printf("%s:%d\n", result.FilePath, errorMatch.LineNumber)
+		}
+	}
+
+	totalErrors := 0
+	for _, result := range results {
+		totalErrors += len(result.Errors)
+	}
+	fmt.Printf("\nTotal: %d non-smarterr error patterns found across %d files.\n", totalErrors, len(results))
+}
+
+func fixFiles(results []FileResult) error {
+	for _, result := range results {
+		hasBareReturns := false
+		for _, errorMatch := range result.Errors {
+			if errorMatch.Pattern == "bare_return_nil_err" {
+				hasBareReturns = true
+				break
+			}
+		}
+
+		if !hasBareReturns {
+			continue
+		}
+
+		err := fixBareReturnsInFile(result.FilePath)
+		if err != nil {
+			return fmt.Errorf("failed to fix file %s: %v", result.FilePath, err)
+		}
+	}
+	return nil
+}
+
+func fixBareReturnsInFile(filePath string) error {
+	content, err := os.ReadFile(filePath)
+	if err != nil {
+		return err
+	}
+
+	lines := strings.Split(string(content), "\n")
+	modified := false
+	needsSmarterrImport := false
+
+	hasSmarterrImport := false
+	for _, line := range lines {
+		if strings.Contains(line, `"github.com/YakDriver/smarterr"`) {
+			hasSmarterrImport = true
+			break
+		}
+	}
+
+	bareReturnRegex := regexp.MustCompile(`^(\s*)return\s+nil\s*,\s*err\s*$`)
+	for i, line := range lines {
+		if bareReturnRegex.MatchString(line) {
+			matches := bareReturnRegex.FindStringSubmatch(line)
+			if len(matches) > 1 {
+				indent := matches[1]
+				lines[i] = indent + "return nil, smarterr.NewError(err)"
+				modified = true
+				needsSmarterrImport = true
+			}
+		}
+	}
+
+	if needsSmarterrImport && !hasSmarterrImport {
+		lines = addSmarterrImport(lines)
+		modified = true
+	}
+
+	if modified {
+		newContent := strings.Join(lines, "\n")
+		err = os.WriteFile(filePath, []byte(newContent), 0644)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func addSmarterrImport(lines []string) []string {
+	for i, line := range lines {
+		if strings.HasPrefix(strings.TrimSpace(line), "import (") {
+			lines = append(lines[:i+1], append([]string{"\t\"github.com/YakDriver/smarterr\""}, lines[i+1:]...)...)
+			return lines
+		} else if strings.HasPrefix(strings.TrimSpace(line), "import ") && strings.Contains(line, "\"") {
+			importLine := strings.TrimSpace(line)
+			existingImport := strings.TrimPrefix(importLine, "import ")
+
+			lines[i] = "import ("
+			lines = append(lines[:i+1], append([]string{
+				"\t" + existingImport,
+				"\t\"github.com/YakDriver/smarterr\"",
+				")",
+			}, lines[i+1:]...)...)
+			return lines
+		}
+	}
+	return lines
+}
```

Comment:
Could the `make fix-imports` target be paired with this tool to handle cleaning up imports?